lib/http/features/auto_inflate.rb

```diff
@@ -17,15 +17,15 @@ def wrap_response(response)
           :headers       => response.headers,
           :proxy_headers => response.proxy_headers,
           :connection    => response.connection,
-          :body          => stream_for(response.connection),
+          :body          => stream_for(response.connection, response.body.encoding),
           :request       => response.request
         }
 
         Response.new(options)
       end
 
-      def stream_for(connection)
-        Response::Body.new(Response::Inflater.new(connection))
+      def stream_for(connection, encoding)
```

Comment:
This would be a breaking change to a public API, however you could do:

```suggestion
      def stream_for(connection, encoding: Encoding::BINARY)
```

Reply:
Fixed!